.github/workflows/ci-run.yml

```
@@ -37,69 +43,478 @@ jobs:
+                      command: cargo test --locked --lib --bins --verbose
+        steps:
+            - uses: actions/checkout@34e114876b0b11c390a56381ad16ebd13914f8d5 # v4
+            - uses: dtolnay/rust-toolchain@631a55b12751854ce901bb631d5902ceb48146f7 # stable
+              with:
+                  toolchain: 1.92.0
+            - uses: Swatinem/rust-cache@779680da715d629ac1d338a641029a2f4372abb5 # v3
+              with:
+                  prefix-key: ci-run-cross-vm-${{ matrix.name }}
+                  cache-bin: false
+            - name: Build and test on VM
+              shell: ${{ matrix.shell }}
```

actionlint (1.7.11)

[error] context "matrix" is not allowed here. no context is available here. see https://docs.github.com/en/actions/learn-github-actions/contexts#context-availability for more details

(expression)